pkg/utils/retry/error.go

```diff
@@ -0,0 +1,26 @@
+package retry
+
+// PermanentError signals that the operation should not be retried.
+type PermanentError struct {
+	Err error
+}
+
+// Permanent wraps the given err in a *PermanentError.
+func Permanent(err error) error {
```

Comment:
Semantically I think it's confusing that this returns nil if the error is nil; are we ever likely to call this with a nil error? (I would imagine not since you'll only be wrapping something you know to be an error)

Reply:
moved the nil check to before String